test.py

```diff
@@ -1,124 +1,527 @@
+
+
+    def test_set_assignee(self):
+        gh_provider = GithubApiProvider("jdm", "a453b3923e893f0383cd2893f...", self.owner, self.repo)
+        issue_num = 1
+        assignee = "jdm"
+        gh_provider.api_req = Mock(return_value = {"header":"", "body":assignee})
+
+        self.assertEquals(assignee, gh_provider.set_assignee(assignee, issue_num))
+        gh_provider.api_req.assert_called_with("PATCH", GithubApiProvider.issue_url % (self.owner, self.repo, issue_num), {"assignee":assignee})
+
+@patch('travisciapiprovider.urllib.urlopen')
+class TestTravisCiApiProvider(unittest.TestCase):
+    class FakeFile():
+        def __init__(self, contents):
+            self.contents = contents
+
+
+        def read(self):
+            return self.contents
+
+
+    def setUp(self):
+        self.travis = TravisCiApiProvider()
+
+
+    def test_get_build(self, urlopen_mock):
+        urlopen_mock.return_value = self.FakeFile(json.dumps({"Something":"here"}))
+        self.travis.get_build(1)
+        urlopen_mock.assert_called_with(TravisCiApiProvider.build_url.format(build_id=1))
+
+
+    def test_get_log(self, urlopen_mock):
+        job_id = 1
+        urlopen_mock.return_value = self.FakeFile(open('resources/single-line-comment.log').read())
+        self.travis.get_log({"matrix":[{'id':job_id}]})
+        urlopen_mock.assert_called_with(TravisCiApiProvider.log_url.format(job_id=job_id))
+
+
+    def test_get_pull_request_number(self, urlopen_mock):
+        pr_num = 1234
+        self.assertEquals(1234, self.travis.get_pull_request_number({"compare_url":"https://github.com/servo/servo/{}".format(pr_num)}))
+
+
+class TestErrorLogParser(unittest.TestCase):
+    def setUp(self):
+        self.log_parser = ErrorLogParser()
+
+
+    def test_parse_log(self):
+        with self.assertRaises(NotImplementedError):
+            self.log_parser.parse_log("log", "regex")
+
+
+class TestServoErrorLogParser(unittest.TestCase):
+    def setUp(self):
+        self.error_parser = ServoErrorLogParser()
+        self.multi_log = open('resources/multi-line-comment.log').read()
+        self.expected_multi_errors = error_sample
+        self.single_log = open('resources/single-line-comment.log').read()
+        self.expected_single_errors = [
+            {
+                'body': 'missing space before {', 
+                'position': 49, 
+                'path': './components/plugins/lints/sorter.rs'
+            }
+        ]
+
+
+    def test_parse_errors(self):
+        self.assertEqual(self.expected_multi_errors, list(self.error_parser.parse_log(self.multi_log)))
+        self.assertEqual(self.expected_single_errors, list(self.error_parser.parse_log(self.single_log)))
+
+
+class TestPayloadHandler(unittest.TestCase):
+    def setUp(self):
+        self.handler = PayloadHandler()
+
+
+    def test_handle_payload(self):
+        with self.assertRaises(NotImplementedError):
+            self.handler.handle_payload("payload")
```

Comment:
Agreed, they may be more confusing than anything. There are other tests like it that I will remove.